pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -422,7 +422,7 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{Requeue: true}, nil
 	}
 
-	if cd.Status.InstallerImage == nil {
+	if !cd.Status.Installed && (cd.Status.InstallerImage == nil || cd.Status.CLIImage == nil) {
```

Comment:
I'm nervous about this line, I don't want to go regenerate a bunch of imageset jobs for clusters that are old, already installed, but don't have a CLIImage set (which they wouldn't because they're old)

Adding the Installed guard is meant to address this.

Otherwise this *should* recreate the imageset job due to the code in this function that deletes the job if it's finished. (but we don't have our images set due to this clause)

Reply:
OK. This looks an adequate approach to me. The controller does not need the installer image or the CLI image after the cluster has been installed, so there is no need to resolve the images at that point.